front/src/applications/editor/tools/utils.ts

```diff
@@ -102,8 +105,20 @@ export function selectEntity(
   // create the next list of layer
   const nextLayers = new Set(actualLayers);
 
+  let neededLayers = EDITOAST_TO_LAYER_DICT[entities[0].objType as EditoastType];
+
+  if (entities.length > 1) {
+    const neededLayersWithDuplicates = entities.map(
+      (entity) => EDITOAST_TO_LAYER_DICT[entity.objType as EditoastType][0]
+    );
+    neededLayers = uniq(neededLayersWithDuplicates);
+  }
```

Comment:
Maybe it would be better like this :
```ts
  const layers = entities.reduce(
    (result, entity) => {
      if (!result.objectTypes.includes(entity.objType as EditoastType)) {
        result.objectTypes.push(entity.objType as EditoastType);

        EDITOAST_TO_LAYER_DICT[entity.objType as EditoastType].forEach((layer) => {
          if (!result.neededLayers.includes(layer)) result.neededLayers.push(layer);
        });
      }
      return result;
    },
    { objectTypes: [] as EditoastType[], neededLayers: [] as string[] }
  );
```

Reply:
Done